apps/SparseMatrix/dense_matrix.cuh

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <iostream>
+#include <vector>
+#include "rxmesh/attribute.h"
+#include "rxmesh/context.h"
+#include "rxmesh/types.h"
+
+namespace rxmesh {
+
+template <typename T, typename IndexT = int>
+struct DenseMatInfo
+{
+    DenseMatInfo(IndexT row_size, IndexT col_size)
+        : m_row_size(row_size), m_col_size(col_size)
+    {
+        cudaMalloc((void**)&m_d_val, bytes());
+        m_ld = m_col_size; // col major for now
+    }
+
+    void set_ones()
+    {
+        std::vector<T> init_tmp_arr(m_row_size * m_col_size, 1);
+        CUDA_ERROR(cudaMemcpy(m_d_val,
+                              init_tmp_arr.data(),
+                              bytes() * sizeof(T),
+                              cudaMemcpyHostToDevice));
+    }
+
+    // row major 
+    // __host__ __device__ T& operator()(const uint32_t row,
+    //                                   const uint32_t col)
+    // {
+    //     return m_d_val[row * m_col_size + col];
+    // }
+
+    // __host__ __device__ T& operator()(const uint32_t row,
+    //                                   const uint32_t col) const
+    // {
+    //     return m_d_val[row * m_col_size + col];
+    // }
+
+    __host__ __device__ T& operator()(const uint32_t row, const uint32_t col)
+    {
+        return m_d_val[col * m_row_size + row];  // pitch & stride
+    }
+
+    __host__ __device__ T& operator()(const uint32_t row,
+                                      const uint32_t col) const
+    {
+        return m_d_val[col * m_row_size + row];
+    }
```

Comment:
I guess these two functions should only be called from the device since `m_d_val` is allocated on the device only. So, we can drop `__host__`